extension/src/fileSystem/index.ts

```diff
@@ -63,6 +64,35 @@ export const findAbsoluteDvcRootPath = async (
   return [absoluteRoot]
 }
 
+// .git inside a submodule is a file with the following content: `gitdir: ../.git/modules/demo`
+const findDotGitDir = (gitRoot: string) => {
+  const dotGitPath = join(gitRoot, gitPath.DOT_GIT)
+
+  const isSubmodule = lstatSync(dotGitPath).isFile()
+  if (isSubmodule) {
+    const dotGitAsFileContent = readFileSync(dotGitPath, 'utf8')
+    const gitDirPrefix = 'gitdir: '
+    const gitDirLine = dotGitAsFileContent
+      .split(/\r?\n/)
+      .find(line => line.indexOf(gitDirPrefix) === 0)
+    return resolve(gitRoot, ...(gitDirLine?.slice(8).split('/') || []))
+  }
+  return dotGitPath
+}
+
+const gitRootGitDir: { [key: string]: string } = {}
+
+export const getGitPath = (gitRoot: string, path: string) => {
```

Comment:
[F] Moved this out of constants as it is reading the file system. We could probably even afford to drop `gitRootGitDir` if we wanted to.